components/script/dom/eventsource.rs

```diff
@@ -512,20 +507,21 @@ impl Runnable for ReestablishConnectionRunnable {
     }
 }
 
-pub struct RefetchRequestRunnable {
+#[derive(JSTraceable, HeapSizeOf)]
+pub struct EventSourceTimeoutCallback {
+    #[ignore_heap_size_of = "Because it is non-owning"]
     event_source: Trusted<EventSource>,
-    gen_id: GenerationId,
-
-    event_type: String,
-    data: String,
-    last_event_id: String,
+    #[ignore_heap_size_of = "Because it is non-owning"]
+    action_sender: ipc::IpcSender<FetchResponseMsg>,
+    #[ignore_heap_size_of = "Defined in std"]
+    done_receiver: Receiver<()>
 }
 
-impl Runnable for RefetchRequestRunnable {
-    fn name(&self) -> &'static str { "EventSource RefetchRequestRunnable" }
-
+impl EventSourceTimeoutCallback {
     // https://html.spec.whatwg.org/multipage/#reestablish-the-connection
-    fn handler(self: Box<RefetchRequestRunnable>) {
+    pub fn invoke(self) {
+        // Step 4
+        let _ = self.done_receiver.recv();
```

Comment:
Hmm... so perhaps the timer should be started in the ReestablishConnectionRunnable instead?